### PR TITLE
Issue #1835: Improve tests speed by reusing setUp structure for each test

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/1835.

Replaces https://github.com/backdrop/backdrop/pull/1365.

### CODE REVIEW

File: core/scripts/run-tests.sh [hunk 1]
```diff
@@ -282,6 +323,7 @@ function simpletest_script_parse_args() {
     'test-id' => 0,
     'execute-test' => '',
     'xml' => '',
+    'cache' => FALSE,
```
Comment (fabianx-ai) [inline core/scripts/run-tests.sh:326]: Should have probably moved 'cache' above the // used internally section.

File: core/scripts/run-tests.sh [hunk 2]
```diff
@@ -875,3 +930,65 @@ function simpletest_script_print_alternatives($string, $array, $degree = 4) {
...
+
+  foreach($profiles as $profile) {
+    // Delete temporary files directory.
+    file_unmanaged_delete_recursive($file_public_path . '/simpletest/simpletest_cache_' . $profile);
+    backdrop_set_message(t('Cleared cache folder for profile !profile.', array('!profile' => $profile)));
+  }
+}
+
+/**
+ * Removed profile cached tables from the database.
```
Comment (fabianx-ai) [inline core/scripts/run-tests.sh:973]: Wrong doxygen

**Comment by Gormartsen** [comment]
@LionsAd can you open issue please? I was passing by closed issue and just found your comments...

**Comment by fabianx-ai** [comment]
@Gormartsen Unfortunately I won't have the time for that, it was just a quick drive-by-review.

@Gormartsen @quicksketch:

This is what I created out of this patch:

   https://www.drupal.org/node/2759197#comment-11358555

Thank you very much for pointing me to it. Should have done that years ago already.

It should be easily back-portable - if you are interested.

Features:

--cache-modules mode for local development. (It rocks to run tests locally in 1-2s instead of 9-13s)

--cache mode for generic speedup, all profiles work, no more pre-seeding is needed.

Cache is kept automatically until cleaned with --clean => again useful for local development.

**Comment by Gormartsen** [comment]
@LionsAd well done! Btw gratz with promotion !

**Pull request merged**